gateway/mw_external_oauth_test.go

```diff
@@ -0,0 +1,369 @@
+		spec := BuildAPI(func(spec *APISpec) {
+			spec.UseKeylessAccess = false
+			spec.ExternalOAuth = apidef.ExternalOAuth{
+				Enabled: true,
+				Providers: []apidef.Provider{
+					{
+						JWT: apidef.JWTValidation{
+							Enabled:       true,
+							SigningMethod: HMACSign,
+							Source:        base64.StdEncoding.EncodeToString([]byte(jwtSecret)),
+						},
+					},
+				},
+			}
+			spec.Proxy.ListenPath = "/"
+		})[0]
+
+		_ = ts.Gw.LoadAPI(spec)
+
+		t.Run("base64 encoded static secret - success", func(t *testing.T) {
+			jwtToken := createJWKTokenHMAC(func(t *jwt.Token) {
+				t.Claims.(jwt.MapClaims)["sub"] = "bar"
+				t.Claims.(jwt.MapClaims)["exp"] = time.Now().Add(time.Hour * 72).Unix()
+			})
+
+			authHeaders := map[string]string{"authorization": jwtToken}
+			_, _ = ts.Run(t, test.TestCase{
+				Headers: authHeaders, Code: http.StatusOK,
+			})
+		})
+
+		t.Run("base64 encoded static secret - failure", func(t *testing.T) {
+			jwtToken := createJWKTokenHMAC(func(t *jwt.Token) {
+				t.Claims.(jwt.MapClaims)["sub"] = "bar"
+				t.Claims.(jwt.MapClaims)["exp"] = time.Now().Add(time.Hour * 72).Unix()
+			})
+
+			t.Run("invalid signature", func(t *testing.T) {
+				token := jwtToken + "blah"
+				authHeaders := map[string]string{"authorization": token}
+				_, _ = ts.Run(t, test.TestCase{
+					Headers: authHeaders, Code: http.StatusUnauthorized,
+				})
+			})
+
+			t.Run("invalid token", func(t *testing.T) {
+				token := "blah"
+				authHeaders := map[string]string{"authorization": token}
+				_, _ = ts.Run(t, test.TestCase{
+					Headers: authHeaders, Code: http.StatusUnauthorized,
+				})
+			})
+
+			t.Run("configured HMAC, signed RSA", func(t *testing.T) {
+				token := CreateJWKToken(func(t *jwt.Token) {
+					t.Claims.(jwt.MapClaims)["foo"] = "bar"
+					t.Claims.(jwt.MapClaims)["user_id"] = "user"
+				})
+
+				authHeaders := map[string]string{"authorization": token}
+				_, _ = ts.Run(t, test.TestCase{
+					Headers: authHeaders, Code: http.StatusInternalServerError,
+				})
+			})
+
+			t.Run("identityBaseField is not set and sub is not present", func(t *testing.T) {
+				jwtToken := createJWKTokenHMAC(func(t *jwt.Token) {
+					t.Claims.(jwt.MapClaims)["exp"] = time.Now().Add(time.Hour * 72).Unix()
+				})
+
+				authHeaders := map[string]string{"authorization": jwtToken}
+				_, _ = ts.Run(t, test.TestCase{
+					Headers: authHeaders, Code: http.StatusInternalServerError,
+					BodyMatch: ErrTokenValidationFailed.Error(),
+				})
+			})
+
+			t.Run("invalid base64 encoded secret", func(t *testing.T) {
+				spec.ExternalOAuth.Providers[0].JWT.Source = "invalid-secret"
+				_ = ts.Gw.LoadAPI(spec)
+
+				authHeaders := map[string]string{"authorization": jwtToken}
+				_, _ = ts.Run(t, test.TestCase{
+					Headers: authHeaders, Code: http.StatusInternalServerError,
+					BodyMatch: ErrTokenValidationFailed.Error(),
+				})
+			})
+
+		})
+	})
+
+	t.Run("time validation", func(t *testing.T) {
```

Comment:
Why didn't we write this one separately working just with `timeValidateJWTClaims` ?